docs/contributor/updateability.md

```diff
@@ -0,0 +1,349 @@
+# Upgrading MicroShift
+
+Following document is meant to be a contributor's introduction
+to the feature of MicroShift's updateability.
+
+## On-disk artifacts
+
+New on-disk artifacts were added:
+- `/var/lib/microshift-backups/`
+    - `health.json`
+    - backup directories
+- `/var/lib/microshift/version`
+
+### `/var/lib/microshift-backups/`
+
+New, additional directory for keeping health and backup related artifacts.
+For ostree systems it's a non-configurable directory where backups are created.
+For non-ostree systems it's a default (configurable) directory for backups.
+
+#### `health.json` [ostree-only]
```

Comment:
Greenboot runs on systems without ostree. Automated rollback is only supported with ostree, but some of the other greenboot features, especially reporting system health holistically, are supported. So, this file might exist on a regular system, too.

Reply:
We don't create it on non-ostree ([ref](https://github.com/openshift/microshift/blob/72e8e0aab81342e1db9f773dc5f1d67f0407540c/packaging/greenboot/microshift_set_healthy.sh#L13-L16)):
```sh
if [ ! -f /run/ostree-booted ]; then
    echo "System is not booted with ostree"
    exit 0
fi
```